models/user.js

```diff
@@ -16,6 +18,49 @@ const UserSchema = new Schema({
   }],
 })
 
+UserSchema.pre('save', function hashPassword(next) {
+  async.waterfall(
```

Comment:
Why don't we just call bcypt.hash then assign the hash to password? What happen in there. I'm just asking, not blame.

Reply:
Got it!